packages/core/src/transports/multiplexed.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+  BaseTransportOptions,
+  Envelope,
+  EnvelopeItemType,
+  Event,
+  EventItem,
+  Transport,
+  TransportMakeRequestResponse,
+} from '@sentry/types';
+import { dsnFromString, forEachEnvelopeItem } from '@sentry/utils';
+
+import { getEnvelopeEndpointWithUrlEncodedAuth } from '../api';
+
+interface MatchParam {
+  /** The envelope to be sent */
+  envelope: Envelope;
+  /**
+   * A function that returns an event from the envelope if one exists
+   * @param types Defaults to ['event', 'transaction', 'profile', 'replay_event']
+   */
+  getEvent(...types: EnvelopeItemType[]): Event | undefined;
```

Comment:
m: Can we make this an array argument instead, so `getEvent(types: EnvelopeItemType[])`? This holds the door open for us potentially adding more stuff here in the future, and also IMHO is easily explainable as well?